packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -470,7 +470,7 @@ class ButtonThemeData extends Diagnosticable {
 
   Color _getDisabledColor(MaterialButton button) {
     return getBrightness(button) == Brightness.dark
-      ? colorScheme.onSurface.withOpacity(0.30)  // default == Colors.white30
+      ? colorScheme.onSurface.withOpacity(0.38)  // default == Colors.white38
```

Comment:
Both conditions in are now the same here, so the conditional is unnecessary

Reply:
Made the fix, as well as updated the API docs since I missed updating those as well